tests/v2/validation/rbac/rbac.go

```diff
@@ -160,9 +162,9 @@ func createDeploymentAndWait(steveclient *v1.Client, client *rancher.Client, clu
 		if *deployment.Spec.Replicas == deployment.Status.AvailableReplicas {
 			return true, nil
 		}
-		return false, nil
+		return false, err
```

Comment:
Are you sure you want to return an error here? If there's an error the poll will stop and be reported as a failure.

Reply:
yeah so, the reason why I added this err was in the L156, when I deploy something with restricted PSAs enabled on the namespace, the deployment should fail and the error message needs to be returned. If I make the err nil in the return value, the forbidden error that I get from the failed deployment is not returned and the validations in the tests/v2/validation/rbac/rbac_psa_test.go on line L254 fail as an error is expected there.

Reply:
Will this affect our other rbac tests? What if the availableReplicas don't match the replicas on the first loop?

Reply:
oh wait. sorry my bad! I thought this was outside of the polling function. Actually this is not necessary. The one in L167 is the one I wanted. this was a typo on my part. thanks for pointing that out.